include/deal.II/boost_adaptors/segment.h

```diff
@@ -22,7 +22,7 @@
 
 #include <deal.II/boost_adaptors/point.h>
 
-#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/segment.hpp>
 
```

Comment:
This should be guarded as well, I think.

Reply:
I can do that but I didn't need it to compile warning-free with `gcc-9.3`.